src/main/java/trainingportal/controller/CourseController.java

```diff
@@ -1,85 +1,89 @@
 package trainingportal.controller;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.servlet.ModelAndView;
 import org.springframework.web.servlet.mvc.support.RedirectAttributes;
-import trainingportal.dao.CourseDAOImpl;
 import trainingportal.model.Course;
+import trainingportal.service.CourseServiceImpl;
 
 import java.util.List;
 
 @Controller
 public class CourseController {
-    @Autowired
-    private CourseDAOImpl courseDAO;
-
-    @RequestMapping(value = "/course_create", method = RequestMethod.GET)
-    public ModelAndView showAllCourses(ModelAndView model, Long Id) {
-//    public String showAllCourses(Model model) {
-        List<Course> course = courseDAO.getCoursAll();
-//        model.addAttribute("courseAll", list);
-        model.addObject("courseAll", course);
-        model.setViewName("CourseCreator/course_create");
-        return model;
+
+    private final CourseServiceImpl courseService;
```

Comment:
Add ```@Autowire``` annotation. Don't use constructor b/c It is bad practice.
Hint:
```@Autowired
    private UserService trainerService;```

Reply:
already change still dont work
   @Autowired
     CourseServiceImpl courseService;

Reply:
> Impl
You need to write:
```
CourseService courseService;
```
Why don't you use my example with TrainerService?

Reply:
write, reboot, still dont work

Reply:
and if it will be important, my edit and other stuff wouldn't work either